src/screens/LoginScreen.js

```diff
@@ -1,50 +1,46 @@
 // 3rd part libraries - core
+import {AppLoading} from 'expo';
 import React, {Component} from 'react';
-import {View, Text, Keyboard} from 'react-native';
+import {View, Text, AsyncStorage} from 'react-native';
 import {connect} from 'react-redux';
-import { Button, FormInput, FormLabel } from 'react-native-elements'
+import {Button, Card} from 'react-native-elements'
 // 3rd party libraries - additional
+import _ from 'lodash';
 
 // actions
-import {emailChanged, passwordChanged, loginUser} from '../actions/index';
+import {checkAndSetToken, setGuestToken, logUserIntoFacebook} from '../actions/index';
 
 // language and styles
 import {textDark, lightwhiteblue, facebookBlue, grayblue, navyBlue} from '../styles/index';
 import {login} from '../locale.en';
+import {navKeys} from '../constants';
+
+const {campo_libre, tagline, login_as_guest, login_with_facebook} = login;
 
 
 // our components - core
 // our components - additional
-import {  Input, Card, CardSection, PasswordInput} from '../components/common/index';
-import ListScreen from "../components/SearchList";
-import MapScreen from "../components/SearchMap";
-
 
 class LoginScreen extends Component {
-    state = {
-        hidden: true
-    };
+
+    componentWillMount() {
+        this.props.checkAndSetToken();
+    }
 
     componentWillReceiveProps(nextProps) {
         const {token, navigation: {navigate}} = nextProps;
+
         if (token) {
-            Keyboard.dismiss();
-            navigate('map');
+            navigate(navKeys.SEARCH);
         }
     }
```

Comment:
When props come back (like when we successfully log in, or if `checkAndSetToken` found and set a token on app launch), check for a token, and if there is one then redirect to the `searchScreen` since we're logged in.